src/autoscaler/scalingengine/schedule/schedule_test.go

```diff
@@ -0,0 +1,240 @@
+				err = schedules.SetActiveSchedule("an-app-id", activeSchedule)
+				Expect(err).NotTo(HaveOccurred())
+				policyDB.GetAppPolicyReturns(&models.ScalingPolicy{}, nil)
+			})
+
+			It("removes the active schedule", func() {
+				Expect(err).NotTo(HaveOccurred())
+				_, exists := schedules.GetActiveSchedule("an-app-id")
+				Expect(exists).To(BeFalse())
+			})
+		})
+
+		Context("when  active schedule does not exist", func() {
+			BeforeEach(func() {
+				policyDB.GetAppPolicyReturns(&models.ScalingPolicy{}, nil)
+			})
+
+			It("logs the information", func() {
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(buffer).Should(gbytes.Say("remove-active-schedule"))
+				Eventually(buffer).Should(gbytes.Say("active schedule does not exist"))
+			})
+		})
+
+		Context("when active schedule id does not match", func() {
+			BeforeEach(func() {
+				err = schedules.SetActiveSchedule("an-app-id", ActiveSchedule{
+					ScheduleId: "current-schedule-id",
+				})
+				Expect(err).NotTo(HaveOccurred())
+				policyDB.GetAppPolicyReturns(&models.ScalingPolicy{}, nil)
+			})
+
+			It("logs the information and removes the app schedule", func() {
+				Expect(err).NotTo(HaveOccurred())
+				_, exists := schedules.GetActiveSchedule("an-app-id")
+				Expect(exists).To(BeFalse())
+
+				Eventually(buffer).Should(gbytes.Say("remove-active-schedule"))
+				Eventually(buffer).Should(gbytes.Say("schedule id does not match"))
+			})
+		})
+
+		Context("when current instance number is in the range of default policy", func() {
+			BeforeEach(func() {
+				cfc.GetAppInstancesReturns(3, nil)
+				policyDB.GetAppPolicyReturns(&models.ScalingPolicy{InstanceMin: 1, InstanceMax: 5}, nil)
+			})
+
+			It("does not change the instance number", func() {
+				Expect(cfc.SetAppInstancesCallCount()).To(Equal(0))
+			})
+		})
+
+		Context("when current instance number is less than default instance-min-count", func() {
+			BeforeEach(func() {
+				cfc.GetAppInstancesReturns(1, nil)
+				policyDB.GetAppPolicyReturns(&models.ScalingPolicy{InstanceMin: 3, InstanceMax: 6}, nil)
+			})
+
+			It("changes the instance number to instance-min-count", func() {
+				appId, instances := cfc.SetAppInstancesArgsForCall(0)
+				Expect(appId).To(Equal("an-app-id"))
+				Expect(instances).To(Equal(3))
+			})
+		})
+
+		Context("when current instance number is greater than default instance-max-count", func() {
+			BeforeEach(func() {
+				cfc.GetAppInstancesReturns(8, nil)
+				policyDB.GetAppPolicyReturns(&models.ScalingPolicy{InstanceMin: 3, InstanceMax: 6}, nil)
+			})
+
+			It("changes the instance number to instance-max-count", func() {
+				appId, instances := cfc.SetAppInstancesArgsForCall(0)
+				Expect(appId).To(Equal("an-app-id"))
+				Expect(instances).To(Equal(6))
+			})
+		})
+
+		Context("when fails to get app policy", func() {
+			BeforeEach(func() {
+				policyDB.GetAppPolicyReturns(nil, errors.New("an error"))
+			})
+
+			It("should removes the schedule and error", func() {
```

Comment:
This is very strange. If I report an error, I would assume the remove did not happen.
If we are going to remove regardless, then there is no error to report.